cpp/src/table/row_operators.cu

```diff
@@ -150,6 +247,8 @@ auto decompose_structs(table_view table,
                          std::move(verticalized_col_depths));
 }
 
+using column_checker_fn_t = std::function<void(column_view const&)>;
```

Comment:
What's the benefit of doing this instead of just defining `auto` lambdas?

Reply:
auto lambdas can't recurse